src/ReactWrapper.jsx

```diff
@@ -208,7 +208,10 @@ class ReactWrapper {
       throw new Error('ReactWrapper::unmount() can only be called on the root');
     }
     this.single('unmount', () => {
-      this.component.setState({ mount: false });
+      this.component.setState({
+        mount: false,
+        lastRender: this.component._reactInternalInstance._renderedComponent.getPublicInstance(),
```

Comment:
I gave that a shot and it's not working. What I think is happening is that `setState()` is triggering a rerender which is replacing `_reactInternalInstance`. I can't set the property immediately after  `setState()` because it's asynchronous, and apparently if a component unmounts callbacks passed to `setState()` don't fire.